lib/karto_sdk/include/karto_sdk/Karto.h

```diff
@@ -5734,6 +5744,17 @@ class LocalizedRangeScan : public LaserRangeScan
    */
   Pose2 m_CorrectedPose;
 
+  /**
+   * Covariance of corrected pose.
+   *
+   *   | var-x   cov-xy    0    |
+   *   |                        |
+   *   | cov-xy  var-y     0    |
+   *   |                        |
+   *   |  0        0    var-yaw |
+   */
+  Matrix3 m_Covariance;
```

Comment:
In this case I'm just using the LocalizedRangeScan object as a way to return the covariance matrix to the caller, but I don't think it needs to be serialized with the map.  Maybe it would be better to just modify the Process() signature to have a return parameter with the matrix?

Reply:
Its an attribute of the scan, so it needs to be stored for multisession processing, this could be important information for multisession mapping https://github.com/SteveMacenski/slam_toolbox/blob/18ccd0b0f15d0b659d336f4739bd014e52663de7/lib/karto_sdk/include/karto_sdk/Karto.h#L5718-L5730

Reply:
If we're going to add attributes, we should make sure the serialized map is complete

Reply:
I'm refactoring it to just return covariance as a output parameter.  I think this information is already stored in the graph as part of the sequential edge.

Reply:
I pushed an update which avoids unnecessarily modifying the LocalizedRangeScan object.  I also added covariance output to the other two modes: PROCESS_FIRST_NODE and PROCESS_NEAR_REGION, which I had missed before.